src/read.c

```diff
@@ -1029,18 +1029,45 @@ display_stinking_cloud_positions(int state)
     }
 }
 
+/* getobj callback for armor object to enchant */
+static int
+armor_enchant_ok(struct obj* obj)
+{
+    if (!obj)
+        return GETOBJ_EXCLUDE;
+
+    if (obj->owornmask & W_ARMOR)
+        return GETOBJ_SUGGEST;
+
+    return GETOBJ_EXCLUDE_INACCESS;
```

Comment:
The way `GETOBJ_EXCLUDE_INACCESS` is used elsewhere, it implies that the action could apply to this item, were it not for some specific issue (e.g. wearing armor that's already being worn, dipping a ring that's being worn under gloves).  I think it may not be the best choice as the default case for the callback.

This also doesn't distinguish between valid armor that's just not currently being worn (in which case `GETOBJ_EXCLUDE_INACCESS` would make sense) and random items that are not armor at all -- both will return `GETOBJ_EXCLUDE_INACCESS` and so both will produce the "You cannot target armor that is not worn" message.

Reply:
There's a difference when using the other excludes in that it can return the zero object (all values set to null) rather than null itself, but good point on the message

Reply:
Oh, is that different from `GETOBJ_EXCLUDE_SELECTABLE`?  The comment in hack.h says they behave exactly the same except for the message when nothing applies ("You have nothing else to act on" vs "You have nothing to act on").

Reply:
will check it out after work, I can't remember why I had to set it to that